pkg/datastore/firestore/firestore.go

```diff
@@ -241,11 +241,40 @@ func (s *FireStore) Close() error {
 	return s.client.Close()
 }
 
-func (s *FireStore) fetchCursorDocumentSnapshot(ctx context.Context, colName string, opts datastore.ListOptions) (*firestore.DocumentSnapshot, error) {
-	if opts.Cursor == "" {
-		return nil, nil
+func processCursorArg(opts datastore.ListOptions) ([]interface{}, error) {
+	// Decode last object of previous page stored as opts.Cursor in json format.
+	var obj struct {
+		Id        string `json:"id"`
+		UpdatedAt int64  `json:"updated_at"`
```

Comment:
Shouldn't this be `PrimaryOrderFieldName` and `PrimaryOrderFiledValue` instead of fixing with `updatedAt`?

Reply:
Better than that, maybe this should contain a list of pair of `order_field_name` and `order_field_value`.

Reply:
I think this one contains cursor's values we use internally for pagination so I guess should strict it as `UpdatedAt` and `Id` 👀

Reply:
> maybe this should contain a list of pair of order_field_name and order_field_value

I get your point, we're going to support pagination on other fields too 👀

Reply:
Added workaround to support other fields than UpdatedAt and Id as cursor values by this commit https://github.com/pipe-cd/pipe/pull/1864/commits/e75aecb4fe8c49a07590b28d0d8df35f10ae4923 🙏 PTAL when you have time 👀